### PR TITLE
`automanage`: refactoring to use `hashicorp/go-azure-sdk`

### PR DESCRIPTION
**Pull request by tombuildsstuff**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave "+1" or "me too" comments, they generate extra noise for PR followers and do not help prioritize for review

## Description

This PR updates the `automanage` service package to use `hashicorp/go-azure-sdk` rather than `tombuildsstuff/kermit`.

During this migration I've discovered that the `AzureStackHCI Cluster` resource is unexpectedly managing the association between the AzureStackHCI Cluster and the associated AutoManage Configuration, rather than exposing this association as separate resource (as this should be) - as such I've added TODOs for us to fix this in 4.0.

Unfortunately it appears that despite the Service Team committing that there would be no breaking changes to the payload (specifically that any changes would be additive) - there's been a breaking change to the payload since [the original pull request was merged](https://github.com/hashicorp/terraform-provider-azurerm/pull/21490) - given that the `TestAccAutoManageConfigurationProfile_complete` and `TestAccAutoManageConfigurationProfile_update` tests haven't passed for some time.

As such - @liuwuliuyun / @mybayern1974 can you please reach out to the Service Team to find out what's happened here?

### CODE REVIEW

File: internal/services/automanage/migration/configuration_v0_to_v1.go [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+	"github.com/hashicorp/go-azure-sdk/resource-manager/automanage/2022-05-04/configurationprofiles"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+var _ pluginsdk.StateUpgrade = ConfigurationV0ToV1{}
+
+type ConfigurationV0ToV1 struct {
+}
+
+func (ConfigurationV0ToV1) Schema() map[string]*pluginsdk.Schema {
```
Comment (stephybun) [inline internal/services/automanage/migration/configuration_v0_to_v1.go:16]: Per our [state migration guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/guide-state-migrations.md) we shouldn't need to have `ForceNew`, `Default` and `MaxItems` in the schema for state migrations?

File: internal/services/automanage/migration/configuration_v0_to_v1.go [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+
+		// "Antimalware/Enable": boolean, true if block exists
+		// "Antimalware/EnableRealTimeProtection": boolean,
+		// "Antimalware/RunScheduledScan": boolean,
+		// "Antimalware/ScanType": string ("Quick", "Full"),
+		// "Antimalware/ScanDay": int (0-8) Ex: 0 - daily, 1 - Sunday, 2 - Monday, .... 7 - Saturday, 8 - Disabled,
+		// "Antimalware/ScanTimeInMinutes": int (0 - 1440),
+		// "Antimalware/Exclusions/Extensions": string (extensions separated by semicolon. Ex: ".ext1;.ext2"),
+		// "Antimalware/Exclusions/Paths": string (Paths separated by semicolon. Ex: "c:\excluded-path-1;c:\excluded-path-2"),
+		// "Antimalware/Exclusions/Processes": string (Processes separated by semicolon. Ex: "proc1.exe;proc2.exe"),
```
Comment (stephybun) [inline internal/services/automanage/migration/configuration_v0_to_v1.go:44]: do we need this comment here for the state migration?

File: internal/services/automanage/migration/configuration_v0_to_v1.go [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+
+		// "AzureSecurityBaseline/Enable": boolean, true if block exists
+		// "AzureSecurityBaseline/AssignmentType": string ("ApplyAndAutoCorrect", "ApplyAndMonitor", "Audit", "DeployAndAutoCorrect"),
```
Comment (stephybun) [inline internal/services/automanage/migration/configuration_v0_to_v1.go:102]: same here

File: internal/services/automanage/migration/configuration_v0_to_v1.go [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+		// "Backup/SchedulePolicy/ScheduleRunTimes": list of DateTime,
+		// "Backup/SchedulePolicy/ScheduleRunDays": list of strings (["Sunday", "Monday", "Wednesday", "Thursday", "Friday", "Saturday"]),
+		// "Backup/SchedulePolicy/SchedulePolicyType": string ("SimpleSchedulePolicy"),
+		// "Backup/RetentionPolicy/RetentionPolicyType": string ("LongTermRetentionPolicy"),
+		// "Backup/RetentionPolicy/DailySchedule/RetentionTimes": list of DateTime,
+		// "Backup/RetentionPolicy/DailySchedule/RetentionDuration/Count": int (7 - 9999),
+		// "Backup/RetentionPolicy/DailySchedule/RetentionDuration/DurationType": string ("Days"),
+		// "Backup/RetentionPolicy/WeeklySchedule/RetentionTimes":, list of DateTime
+		// "Backup/RetentionPolicy/WeeklySchedule/RetentionDuration/Count":, int (1 - 5163)
+		// "Backup/RetentionPolicy/WeeklySchedule/RetentionDuration/DurationType": string ("Weeks"),
```
Comment (stephybun) [inline internal/services/automanage/migration/configuration_v0_to_v1.go:132]: same here

**Comment by liuwuliuyun** [comment]
@tombuildsstuff the tests failure are probably due to the lack of permission from test subscription. You could enable this feature flag by following steps in https://github.com/hashicorp/terraform-provider-azurerm/issues/22799.

Local test on the original code:

```
GOROOT=C:\Program Files\Go #gosetup
GOPATH=C:\Users\yunliu1\go #gosetup
"C:\Program Files\Go\bin\go.exe" test -c -o C:\Users\yunliu1\AppData\Local\JetBrains\GoLand2023.3\tmp\GoLand\___TestAccAutoManageConfigurationProfile_complete_in_github_com_hashicorp_terraform_provider_azurerm_internal_services_automanage.test.exe github.com/hashicorp/terraform-provider-azurerm/internal/services/automanage #gosetup
"C:\Program Files\Go\bin\go.exe" tool test2json -t C:\Users\yunliu1\AppData\Local\JetBrains\GoLand2023.3\tmp\GoLand\___TestAccAutoManageConfigurationProfile_complete_in_github_com_hashicorp_terraform_provider_azurerm_internal_services_automanage.test.exe -test.v -test.paniconexit0 -test.run ^\QTestAccAutoManageConfigurationProfile_complete\E$ #gosetup
=== RUN   TestAccAutoManageConfigurationProfile_complete
=== PAUSE TestAccAutoManageConfigurationProfile_complete
=== CONT  TestAccAutoManageConfigurationProfile_complete
--- PASS: TestAccAutoManageConfigurationProfile_complete (138.23s)
PASS


Process finished with the exit code 0
```

**Comment by tombuildsstuff** [comment]
@liuwuliuyun in our case those features aren't registered, so that makes sense - however the API returns this rather cryptic when the feature isn't enabled:

```
 Error: creating Configuration Profile (Subscription: "XXX"
         Resource Group Name: "acctest-rg-240319093707573983"
         Configuration Profile Name: "acctest-amcp-240319093707573983"): unexpected status 400 with error: > InvalidConfigurationName: The following invalid configuration names were found:  <Alerts/AutomanageStatusChanges/Enable>
```

Whereas previously (from #22799) it returned the much more helpful:

```
Error: creating Automanage Configuration: (Configuration Profile Name "Example" / Resource Group "Example"): automanage.ConfigurationProfilesClient#CreateOrUpdate: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="InvalidConfigurationValue" Message="The feature flag <Microsoft.Automanage/AutomanageAlertsEnabled> must enabled on subscription: <subscription>, in order to use properties: <Alerts/AutomanageStatusChanges/Enable>"
```

As such [given the documentation doesn't show this](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/automanage_configuration#status_change_alert_enabled) - would you mind adding documentation to highlight that this is needed?

Thanks!

File: internal/services/automanage/migration/configuration_v0_to_v1.go (hunk 2 above)
Comment (tombuildsstuff) [reply internal/services/automanage/migration/configuration_v0_to_v1.go]: Good shout, removed.

**Review by stephybun — APPROVED** [review]
There are still `MaxItems` attributes in the schema for the state migration but otherwise LGTM 👍

**Pull request merged**